client/app/pages/dashboards/hooks/utils.js

```diff
@@ -0,0 +1,18 @@
+export function getAllowedWidgetsForCurrentParam(dashboardParameters, dashboardAllowedWidgets, dashboardAllWidgets) {
+  try {
+    // filter the widgets by keeping only the ones we want
+    for (var param of dashboardParameters) {
+      const paramValue = param.value;
+      // check if we have allowed widgets for this param value
+      if (dashboardAllowedWidgets !== undefined && dashboardAllowedWidgets.hasOwnProperty(paramValue)) {
+        // filter the widgets to process
+        const allowedWidgets = dashboardAllowedWidgets[paramValue];
+        return dashboardAllWidgets.filter(widget => allowedWidgets.includes(widget.visualization.name));
```

Comment:
You're missing the case where the visualization name has more than 1 widget. For example: "heat1;heat2"

Reply:
according to what i got so far, one visualisation is really linked to one query -> to one widget ... but then we can add many visualisations to one widget and give each one a name, in this case our list of allowed widgets can have all of them or some of them or none.. as we want... the current code supports this case. 

It seems that in order to add many visualisations to a dashboard, we add every time a widget and then pick the visualisation that we want ... this is why it is already supported

am I missing something?

Reply:
My point isn't related to Redash, but to remote monitoring. @helenalebreton Can you help to see if this makes sense?

I was thinking that the same visualization could be used in a case of more than 1 group. For example:

We would have a visualization named "heat1;heat2". This is related to the same query, where just the controller ID changes.

Then imagine we have 2 controllers. One in a building with just just one heat circuit, so it would have a group type Heat and number 1. And another building with 2 heating circuits, then it would have two groups with type Heat, one with number 1 and another with number 2.

My expectation then is that the allowed_widgets query would have this data:

|main_parameter|widgets|
|--------|--------|
|123| ['heat1'] |
|456| ['heat1', 'heat2']|

Then, for me, that visualization should work on both controllers. But now that I'm writing it I'm not sure anymore that this is a valid use case.

Reply:
thank you for your explanation , after @helenalebreton [confirmation](https://github.com/metr-systems/redash/pull/3#issuecomment-2312434216) , i updated the code to support a compound visrtualisation [name in this commit](https://github.com/metr-systems/redash/pull/3/commits/38afa1c89eb2276a4c8823adbcef7d8321a60c2e)